File: backend/app/routers/market_data.py

from fastapi import APIRouter,Depends
from sqlalchemy.orm import Session
from app.db.session import get_db
from app.schemas.market_data import MarketDataCreate,MarketDataResponse
from app.services import market_data_service
from app.core.dependencies import require_admin

router=APIRouter(prefix="/market-data", tags=["Market Data"])

@router.post("/", response_model=MarketDataResponse)
def create(data:MarketDataCreate, db:Session=Depends(get_db), user=Depends(require_admin)):
    return market_data_service.create(db,data)

@router.get("/", response_model=list[MarketDataResponse])
def all(db:Session=Depends(get_db)): return market_data_service.get_all(db)

File: vessel-market-frontend/src/api/marketData.ts
import { api } from "./http";
import type { MarketData, MarketDataPayload } from "./types";

export async function getMarketData() {
  const { data } = await api.get<MarketData[]>("/market-data/");
  return data;
}

export async function createMarketData(payload: MarketDataPayload) {
  const { data } = await api.post<MarketData>("/market-data/", payload);
  return data;
}
